src/scripts/blah_common_submit_functions.sh

```diff
@@ -782,13 +784,22 @@ function bls_test_working_dir ()
       rm -f $bls_tmp_file
       exit 1
   fi
+
+  # Ensure local files actually exist. When called before job submission, this prevents
+  # unnecessary churn on the scheduler if the files don't exist.
+  if ! bls_fl_test_exists inputsand ; then
+      echo "Input sandbox file doesn't exist: $bls_fl_test_exists_result" >&2
+      echo Error # for the sake of waiting fgets in blahpd
+      rm -rf $bls_tmp_file
+      exit 1
+  fi
 }
 
 function bls_add_job_wrapper ()
 {
+  bls_test_input_files
   bls_start_job_wrapper >> $bls_tmp_file
   bls_finish_job_wrapper >> $bls_tmp_file
-  bls_test_working_dir
```

Comment:
Note that previously, `bls_test_working_dir` was getting run after `$bls_tmp_file` was written to, and then deleting `$bls_tmp_file` upon error.  Now `bls_test_input_files` is run beforehand, but still removing the file on error.  Not that it's a problem, but does this file get created somewhere else first?  Or is this spot in `bls_add_job_wrapper` the first?

Reply:
See the description of the PR:

```
Use the pre-existing workdir test function, which catches permission
issues with the workdir as well as non-existence. Do this all before
we write out the job wrapper because why bother?
```

Reply:
I guess my main question and point (which i did not find the answer to in the PR description) are:

> does this file \[`$bls_tmp_file`\] get created somewhere else first? Or is this spot in bls_add_job_wrapper the first?

Because:
- if it gets created somewhere else first, then it still makes sense to delete it in `bls_test_input_files`
- if the above two lines in `bls_add_job_wrapper` are the first place where `$bls_tmp_file` is created/written to, then it is superfluous to do `rm -f $bls_tmp_file` in `bls_test_input_files`, as it will never yet exist at that point.

Reply:
Ok, then it makes sense as-is.